Feature/llm integration

ddoguma opened the pull request:
## 작업 내용

### LLM 기반 블로그 콘텐츠 자동 생성 기능 구현

**기능 요약:**
- OpenAI API 연동하여 정책 데이터 → 블로그 콘텐츠 자동 변환
- DB 정제 데이터(`content_data`)를 활용한 프롬프트 생성
- 10개 카테고리별 맞춤 톤앤매너 적용
- 청년 친화적 구어체 ("~해요", "~있어요", "~해보세요")
- SEO 최적화 (롱테일 키워드, 연관 키워드 활용)

---

## 주요 변경사항

### 1. 새로운 API 엔드포인트
- `POST /api/generate-title` - 블로그 제목 생성
- `POST /api/generate-summary` - 블로그 요약 생성  
- `POST /api/generate-blog-content` - 블로그 본문 생성
- `POST /api/generate-full-blog` - 제목+요약+본문 일괄 생성

### 2. 카테고리별 프롬프트 (10개)

### 3. 프로젝트 구조
- `utils/llm_utils.py` - PromptGenerator 클래스
- `routes/prompts.py` - LLM API 라우터
- `prompts_config/` - 프롬프트 템플릿 관리 (JSON)

Commit:
feat(preproc): 하이브리드 카테고리 분류 및 blog_json 구조 개선

- feat(preproc): 정부 원본 분류 + 키워드 기반 하이브리드 분류

- feat(storage): blog_json Flat 구조 개선

- refactor: 10개 청년 친화적 카테고리 확장

- docs: README 업데이트

ddoguma committed:
Update README.md

Commit:
refactor: blog_json 변수명을 content_data로 변경

- build_blog_json() -> build_content_data() 함수명 변경
- 코드 전반에서 blog_json 변수명을 content_data로 개선
- 데이터 용도가 더 명확하게 표현되도록 수정
- DB 스키마(blog_json 컬럼)는 유지

Commit:
refactor: DB 컬럼명 blog_json을 content_data로 변경

- policy_clean 테이블의 blog_json 컬럼을 content_data로 변경
- storage_pg.py의 DDL 및 SQL 쿼리 업데이트
- routes/policies.py의 API 응답 스키마 및 쿼리 업데이트
- 코드와 DB 스키마 일관성 확보

Commit:
feat: documents 필드 간소화 기능 추가

- simplify_documents() 함수 추가로 복잡한 서류 목록을 청년 친화적으로 요약
- 핵심 서류만 추출 (최대 4개 + '등')
- 숫자 기호 및 긴 괄호 설명 제거
- content_data에 간소화 버전(documents)과 원본(documents_full) 모두 보관
- 평균 90%+ 텍스트 축소 효과

Commit:
feat: 데이터 정제 + SEO 최적화 LLM 프롬프트 통합

✨ 주요 변경사항:
- llm-datasample 브랜치에서 프롬프트 구조 가져오기
- SEO 최적화 프롬프트 적용 (하위/롱테일 키워드, 연관 키워드 활용)
- 데이터 정제 기능과 LLM 프롬프트 통합
- utils 패키지 설정 (llm_utils.py + __init__.py)

📝 프롬프트 최적화 내용:
1. blog_content: SEO 전문가 역할, 키워드 전략, 문장 구조 다양화
2. summary: 청년 친화적 언어, Call-to-Action 포함
3. title: 클릭 유도 제목, 이모지, 30자 이내 제한

🎯 프로젝트 목표 반영:
- 청년 정책을 쉽게 이해할 수 있도록 자동 변환
- 분야별 맞춤 톤앤매너 적용
- SEO 키워드 자동 삽입으로 검색 노출 강화

Commit:
fix: prompts 라우터 및 필수 파일 추가

- routes/prompts.py: 프롬프트 생성 API 엔드포인트 추가
- utils/data_utils.py: 더미 데이터 유틸 추가
- main.py: prompts 라우터 등록

이제 Swagger UI에서 프롬프트 관련 엔드포인트를 확인할 수 있습니다:
- POST /api/generate-title
- POST /api/generate-summary
- POST /api/generate-blog-content
- POST /api/generate-full-blog

Commit:
feat: DB 정제 데이터를 LLM 프롬프트에 연결

- prompts.py: 더미 데이터 제거, DB에서 실제 정제 데이터 조회
- llm_utils.py: blog_json → content_data 구조로 변경
- 정제된 데이터 필드 활용 (benefit, who, apply_method, documents)

데이터 흐름: DB → prompts.py → llm_utils.py → OpenAI API

Commit:
fix: blog_content 프롬프트를 기존 버전으로 변경

마크다운 형식(###) 대신 자연스러운 텍스트로 생성되도록 수정

Commit:
feat: 카테고리별 톤앤매너 & 프롬프트 자동 선택 구현

✨ 주요 기능:
1. 10개 카테고리별 톤앤매너 & 이모지 정의
   - category_config.json: 카테고리별 스타일 가이드
   - category_prompts.json: 카테고리별 프롬프트 템플릿

2. 자동 프롬프트 선택 로직
   - llm_utils.py: 정책 카테고리에 따라 적절한 프롬프트 자동 선택
   - _get_category_prompt(): 카테고리 매칭 로직
   - _load_category_prompts(): 카테고리 프롬프트 로드

3. 카테고리별 차별화
   - 취업 지원 💼: 실용적이고 동기부여하는 톤
   - 교육·자격증 📚: 친근하고 격려하는 톤
   - 창업 🚀: 활기차고 도전적인 톤
   - 주거 🏠: 안정감 있고 따뜻한 톤
   - 대출·금융 💰: 신뢰감 있고 명확한 톤
   - 생활비 지원 💸: 부드럽고 친근한 톤
   - 문화·여가 🎨: 밝고 경쾌한 톤
   - 건강·상담 💚: 따뜻하고 위로하는 톤
   - 해외 기회 ✈️: 설레고 희망찬 톤
   - 청년 참여 🗣️: 활동적이고 격려하는 톤

Commit:
fix: 카테고리별 프롬프트에서 마크다운 형식 제거 지시 추가

모든 카테고리의 content_prompt에 '마크다운 형식을 절대 사용하지 말고 일반 텍스트로만 작성' 지시 추가

Commit:
fix: 기본 blog_content 프롬프트에도 마크다운 제거 지시 강화

blog_content_system_prompt.json에 명확한 마크다운 금지 지시 추가

Commit:
fix: 카테고리별 프롬프트 마크다운 금지 지시 강화

모든 카테고리의 content_prompt에서 마크다운 특수 기호 명시적 금지

Commit:
feat: 청년 친화적 구어체 프롬프트로 전면 강화

- 모든 카테고리별 프롬프트에 구어체 지시 추가
- '~해요', '~있어요', '~해보세요' 같은 부드러운 표현 사용
- 딱딱한 행정 용어 대신 일상 언어로 풀어쓰기
- 마크다운 금지 유지, 1., 2., 3. 번호 형식 허용

Commit:
docs: README 업데이트 - LLM 프롬프트 통합 내용 추가

- LLM 기반 블로그 콘텐츠 자동 생성 기능 설명
- 카테고리별 프롬프트 (10개) 및 톤앤매너 정리
- SEO 최적화 프롬프트 설명
- 새로운 API 엔드포인트 목록 및 테스트 예시
- 프로젝트 구조 업데이트 (prompts_config, utils 폴더)
- 환경변수 OPENAI_API_KEY 추가

louis-lemon commented:
LGTM

POST /api/generate-title - 블로그 제목 생성
POST /api/generate-summary - 블로그 요약 생성
POST /api/generate-blog-content - 블로그 본문 생성
POST /api/generate-full-blog - 제목+요약+본문 일괄 생성

rest api 설계할때는 위 방법보다는
`POST /policies/{plcy_no}/content` 하나의 엔드포인트에 query param이나 body로 타입 받아서 처리하는게 좋을거같아요.

ex)
```
POST /api/policies/{plcy_no}/content?type=title
POST /api/policies/{plcy_no}/content?type=summary
POST /api/policies/{plcy_no}/content?type=blog
POST /api/policies/{plcy_no}/content?type=full
```

Commit:
refactor(api): 통합 콘텐츠 생성 엔드포인트로 리팩토링

- POST /api/policies/{plcy_no}/content로 통합
- type 쿼리 파라미터로 title/summary/blog/full 지원
- RESTful 설계 개선 및 코드 중복 제거
- 프론트엔드 API 함수 추가
- README 문서 업데이트
